test/framework/src/main/java/org/elasticsearch/index/mapper/MapperServiceTestCase.java

```diff
@@ -155,7 +161,23 @@ protected final SourceToParse source(CheckedConsumer<XContentBuilder, IOExceptio
      * Merge a new mapping into the one in the provided {@link MapperService}.
      */
     protected final void merge(MapperService mapperService, XContentBuilder mapping) throws IOException {
-        mapperService.merge(null, new CompressedXContent(BytesReference.bytes(mapping)), MapperService.MergeReason.MAPPING_UPDATE);
+        merge(mapperService, MapperService.MergeReason.MAPPING_UPDATE, mapping);
+    }
+
+    protected final void merge(MapperService mapperService, String mapping) throws IOException {
+        mapperService.merge(null, new CompressedXContent(mapping), MapperService.MergeReason.MAPPING_UPDATE);
+    }
+
+    protected final void merge(MapperService mapperService,
```

Comment:
This one might should be private. Or we should document why you'd want to call this one rather than the other one.

Reply:
It's used to test merging index templates, which work slightly differently and need a specific `MergeReason` to be passed to the `merge` method.  I'll add javadoc.